### PR TITLE
Adding inline actions to Async Server Tree.

### PR DESCRIPTION
**Pull request by aasimkhan30**
Adding async server tree actions:

Tree items when hovered/selected
Server groups:
image

Connections: 
When connected
image
When disconnected
image

Tree Nodes:
Items can be contributed using 'inline' group from extensions
Database:
image
Tables folder: (add icon opens table designer)
image
Table (I have hidden the rename action, the edit icon opens table designer)
image
Users: (edit icon is rename action and the flat list icon is the properties action)
image

Regular Tree nodes only have refresh
image

Tree when nothing is selected/hovered
image

### CODE REVIEW

File: src/sql/workbench/contrib/dataExplorer/browser/media/connectionViewletPanel.css [hunk 1]
```diff
@@ -46,21 +46,21 @@
 	color: #ffffff;
 }
 
-.server-explorer-viewlet  .monaco-action-bar .action-label {
+/* .server-explorer-viewlet  .monaco-action-bar .action-label {
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/dataExplorer/browser/media/connectionViewletPanel.css:49]: please delete if not needed.

File: src/sql/workbench/contrib/scripting/browser/scripting.contribution.ts [hunk 2]
```diff
@@ -287,6 +288,27 @@ MenuRegistry.appendMenuItem(MenuId.ObjectExplorerItemContext, {
...
 
+MenuRegistry.appendMenuItem(MenuId.ObjectExplorerItemContext, {
+	group: 'inline',
+	order: 7,
+	command: {
+		id: commands.OE_REFRESH_COMMAND_ID,
+		title: localize('refreshNode', "Refresh"),
+		icon: Codicon.refresh
+	},
+	when: ContextKeyExpr.or(
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/scripting/browser/scripting.contribution.ts:299]: why only do refresh for these? I think Refresh should be available for all node types

**Comment by alanrenmsft** [comment]
I like the new inline actions for table and tables, just curious where is the requirement coming from?

**Comment by aasimkhan30** [comment]
> I like the new inline actions for table and tables, just curious where is the requirement coming from?

@alanrenmsft , it was one of things I had planned for my oe work. I also plan on adding filter icons to it.

File: src/sql/workbench/contrib/scripting/browser/scripting.contribution.ts (hunk 2 above)
Comment (aasimkhan30) [reply src/sql/workbench/contrib/scripting/browser/scripting.contribution.ts:299]: @alanrenmsft , I just copied the action above and changed the group.

**Comment by aasimkhan30** [comment]
Couple of design questions @alanrenmsft  and @kburtram 

Which icon looks better for users properties action?
image
image

Should we have both rename and design table in table design or just design icon (design will just have the edit icon then)?
imageimage

**Review by erinstellato-ms — APPROVED** [review]
I like the wrench for the properties better than the three lines.  Otherwise, I think the change is good - definitely looking forward to giving it a test run to see them in action.

**Comment by alanrenmsft** [comment]
> @alanrenmsft , should I remove the delete icon?

I think it should be there.

**Comment by alanrenmsft** [comment]
@aasimkhan30 the `pen` icon to me is for the edit/properties action. I don't think we have to show the rename action in the inline group: edit, delete, refresh should be good enough.

**Comment by cheenamalhotra** [comment]
When clicking Edit `pen` on connection node - does it prompt rename or edit connection? If it's 'Edit' then I believe it might confuse between Edit action on connection v/s Rename action on other elements. Let's keep it's primary action as 'Edit'. You could trigger 'Edit Data' on table as well..

**Pull request closed without merging**